docs/modules/language-reference/pages/index.adoc

```diff
@@ -3179,6 +3180,157 @@ This section discusses language features that are generally more relevant to tem
 <<blank-identifiers,Blank Identifiers>> +
 <<projects,Projects>>
 
+[[object-instantiation]]
+=== Object Instantiation
+
+Object instantiation in Pkl always <<amending-objects, amends>> _some_ value.
+The `new` keyword is a special case of amending where a contextual default value is amended. 
+In Pkl, objects can be instantiated in three major ways:
+
+* Amending some parent property or value directly.
+* Invoking `new` with explicit type information, which always amends the default value of the specified type.
+* Invoking `new` without type information, which amends a contextually inferred value.
```

Comment:
I think that the concept of amending a parent is pretty different to what the rest of this section covers. I don't know if we should group that together here; I'm thinking it might be better to just talk about `new` specifically.

```suggestion
=== Meaning of `new`

Objects in Pkl always <<amending-objects, amends>> _some_ value.
The `new` keyword is a special case of amending where a contextual value is amended. 
In Pkl, there are two forms of `new` objects:

* `new` with explicit type information, for example, `new Foo {}`.
* `new` without type information, for example, `new {}`.
```